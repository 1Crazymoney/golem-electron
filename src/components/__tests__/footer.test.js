jest.unmock('../Footer')

import React from 'react'
import ReactDOM from 'react-dom'
import TestUtils from 'react-dom/test-utils'
import sinon from 'sinon'
import ConnectedFooter, {Footer} from '../footer'


describe('<Footer/>', () => {

    it('should render a footer component', () => {
        const wrapper = shallow(
            <Footer/>
        )
        expect(wrapper).toMatchSnapshot()
    })

    it('should render a footer component', () => {
        const wrapper = mount(
            <Footer preview={false}/>
        )
        expect(wrapper.ref('previewSwitch').prop('defaultChecked')).toBe(false)
    })

});