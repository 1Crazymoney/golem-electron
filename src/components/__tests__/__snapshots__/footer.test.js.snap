// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should render a footer component 1`] = `
<footer
  className="footer"
>
  <div
    className="info-bar__footer"
  >
    <span
      className="progress-status indicator-status "
    />
    <span />
  </div>
  <div
    className="preview-bar__footer"
  >
    <Tooltip
      align={
        Object {
          "offset": Array [
            5,
            -10,
          ],
        }
      }
      arrowContent={
        <div
          className="rc-tooltip-arrow-inner"
        />
      }
      destroyTooltipOnHide={false}
      mouseEnterDelay={1}
      mouseLeaveDelay={0.1}
      overlay={
        <p>
          Preview
        </p>
      }
      placement="topRight"
      prefixCls="rc-tooltip"
      trigger={
        Array [
          "hover",
        ]
      }
    >
      <div
        className="switch-box switch-box--green"
      >
        <label
          className="switch"
        >
          <input
            aria-label="Show previews"
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="switch-slider round"
          />
        </label>
      </div>
    </Tooltip>
    <Tooltip
      align={
        Object {
          "offset": Array [
            5,
            -10,
          ],
        }
      }
      arrowContent={
        <div
          className="rc-tooltip-arrow-inner"
        />
      }
      destroyTooltipOnHide={false}
      mouseEnterDelay={1}
      mouseLeaveDelay={0.1}
      overlay={
        <p>
          Preview Window
        </p>
      }
      placement="topRight"
      prefixCls="rc-tooltip"
      trigger={
        Array [
          "hover",
        ]
      }
    >
      <span
        aria-label="Open Detailed Preview Window"
        className="button__expand icon-new-window disabled"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      />
    </Tooltip>
  </div>
</footer>
`;
